addons/base/views.py

```diff
@@ -281,100 +252,120 @@ def get_auth(auth, **kwargs):
-                                try:
-                                    metric_class.record_for_preprint(
-                                        preprint=node,
-                                        user=auth.user,
-                                        version=fileversion.identifier if fileversion else None,
-                                        path=path,
-                                    )
-                                except es_exceptions.ConnectionError:
-                                    log_exception()
-        if fileversion and provider_settings:
-            region = fileversion.region
-            credentials = region.waterbutler_credentials
-            waterbutler_settings = fileversion.serialize_waterbutler_settings(
-                node_id=provider_settings.owner._id,
-                root_id=provider_settings.root_node._id,
-            )
-    # If they haven't been set by version region, use the NodeSettings or Preprint directly
-    if not (credentials and waterbutler_settings):
-        credentials = node.serialize_waterbutler_credentials(provider_name)
-        waterbutler_settings = node.serialize_waterbutler_settings(provider_name)
 
+@collect_auth
+def get_auth(auth, **kwargs):
+    waterbutler_data = decrypt_and_decode_jwt_payload()
+
+    action = waterbutler_data['action']
+    permission = get_permission_for_action(action)
+
+    node = get_authenticated_node(waterbutler_data['nid'])
+
+    # Fallback to cookie-based/token authentication if the user is not authenticated yet
+    if not auth.user:
+        cookie = waterbutler_data.get('cookie')
+        if cookie:
+            auth.user = OSFUser.from_cookie(cookie)
+
+        authorization = request.headers.get('Authorization')
+        if authorization and authorization.startswith('Bearer '):
+            auth.user = authenticate_via_oauth_bearer_token(auth, node)
+
+    if not check_node_permission(node, auth, permission):
+        raise HTTPError(http_status.HTTP_403_FORBIDDEN)
+
+    if not check_parent_permission_for_action(node, auth, action):
+        raise HTTPError(http_status.HTTP_403_FORBIDDEN)
+
+    provider = node.get_addon(waterbutler_data['provider_name'])
+    if provider.name == 'osfstorage':
+        fileversion = get_file_version_from_wb(waterbutler_data)
+        credentials = fileversion.region.waterbutler_credentials
+        waterbutler_settings = fileversion.serialize_waterbutler_settings(
+            node_id=provider.owner._id,
+            root_id=provider.root_node._id,
+        )
+        if action == 'render':
+            file_signals.file_viewed.send(user=auth.user, file_version=fileversion)
+        if action == 'download':
+            file_signals.file_downloaded.send(user=auth.user, file_version=fileversion)
+    else:
+        credentials = node.serialize_waterbutler_credentials(provider.short_name)
+        waterbutler_settings = node.serialize_waterbutler_settings(provider.short_name)
```

Comment:
nit: shove these into helper methods as well?
```
if provider.name == 'osfstorage':
  credentials, waterbutler_settings = _get_osfstorage_wb_payload(. . .)
else:
  credentials, waterbutler_settings = _get_addon_wb_payload(. . .)
```
Then there's an easy, out-of-the-way place to handle the GV migration